src/main/java/org/reactome/addlinks/AddLinks.java

```diff
@@ -514,20 +512,82 @@ else if (fileProcessorName.get() instanceof List<?>)
 			}
 			// There is a separate list of reference creators to create UniProt references.
 			else if (this.uniprotReferenceCreators.containsKey(refCreatorName))
 			{
 				UPMappedIdentifiersReferenceCreator refCreator = this.uniprotReferenceCreators.get(refCreatorName);
-				if (refCreator instanceof NCBIGeneBasedReferenceCreator)
-				{
-					((NCBIGeneBasedReferenceCreator) refCreator).setCTDGenes( (Map<String, String>) dbMappings.get(CTD_PROCESSOR) );
-				}
+				refCreator.setCTDGenes( (Map<String, String>) dbMappings.get(CTD_PROCESSOR) );
+
 				sourceReferences = this.getIdentifiersList(refCreator.getSourceRefDB(), refCreator.getClassReferringToRefName());
 				refCreator.createIdentifiers(personID, (Map<String, Map<String, List<String>>>) dbMappings.get(fileProcessorName.get()), sourceReferences);
+				// Now check the links for the reference creator (and sub-creators).
+				for (EntrezGeneBasedReferenceCreator subCreator : refCreator.getSubCreators())
+				{
+					String line = checkLinksForRefCreator(subCreator);
+					Files.write(Paths.get(linkCheckReportName), line.getBytes(), StandardOpenOption.APPEND);
+				}
+				String line = checkLinksForRefCreator(refCreator);
+				Files.write(Paths.get(linkCheckReportName), line.getBytes(), StandardOpenOption.APPEND);
+			}
+		}
+	}
+
+	/**
+	 * @param refCreator
+	 * @return
+	 */
+	private String checkLinksForRefCreator(BatchReferenceCreator<?> refCreator)
+	{
+		String targetRefDB = refCreator.getTargetRefDB();
+		String line = "";
+		if (!targetRefDB.toUpperCase().contains(KEGG) && !targetRefDB.toUpperCase().contains(ENSEMBL))
+		{
+			Optional<GKInstance> refDBInst = LinksToCheckCache.getCache().keySet().stream()
+																.filter( inst -> ((inst.getDisplayName().equals(targetRefDB) && LinksToCheckCache.getRefDBsToCheck().contains(targetRefDB))))
+																.findFirst();
+			if (refDBInst.isPresent())
+			{
+				line = checkLinksForRefDB(new LinkCheckManager(), refDBInst.get());
+			}
+			else
+			{
+				line = targetRefDB + "\tN/A\tN/A\n";
+			}
+			return line;
+		}
+		else if (targetRefDB.toUpperCase().contains(KEGG))
+		{
+			StringBuilder sb = new StringBuilder();
+			Set<GKInstance> refDBInsts = LinksToCheckCache.getCache().keySet().stream().filter(inst -> inst.getDisplayName().toUpperCase().contains(KEGG)).collect(Collectors.toSet());
+			for (GKInstance refDB : refDBInsts)
+			{
+				sb.append(checkLinksForRefDB(new LinkCheckManager(), refDB));
+			}
+			line = sb.toString();
+		}
+		else if (targetRefDB.toUpperCase().contains(ENSEMBL))
+		{
+			StringBuilder sb = new StringBuilder();
+			Set<GKInstance> refDBInsts = LinksToCheckCache.getCache().keySet().stream().filter(inst -> inst.getDisplayName().toUpperCase().contains(ENSEMBL)).collect(Collectors.toSet());
+			for (GKInstance refDB : refDBInsts)
+			{
+				sb.append(checkLinksForRefDB(new LinkCheckManager(), refDB));
 			}
+			line = sb.toString();
 		}
```

Comment:
The logic in the `if` and `else if` blocks are identical except for the String value checked for in the `contains`.  Could the logic be moved to a method that accepts the name of the `targetRefDB` and returns the line of interest as a String?